src/state.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { useModel } from "@anywidget/react";
+import type { AnyModel } from "@anywidget/types";
+import { debounce } from "./util";
+
+const debouncedModelSaveViewState = debounce((model: AnyModel) => {
+  const viewState = model.get("_view_state");
+
+  // transitionInterpolator is sometimes a key in the view state while panning
+  // This is a function object and so can't be serialized via JSON.
+  //
+  // In the future anywidget may support custom serializers for sending data
+  // back from JS to Python. Until then, we need to clean the object ourselves.
+  // Because this is in a debounce it shouldn't often mess with deck's internal
+  // transition state it expects, because hopefully the transition will have
+  // finished in the 300ms that the user has stopped panning.
+  if ("transitionInterpolator" in viewState) {
+    console.debug("Deleting transitionInterpolator!");
+    delete viewState.transitionInterpolator;
+    model.set("_view_state", viewState);
+  }
+
+  model.save_changes();
+}, 300);
```

Comment:
Note that this is very tied to view state, so maybe we shouldn't name the below function  `useModelStateDebounced`